src/main.cpp

```diff
@@ -28,22 +38,29 @@ void setup() {
 }
 
 void loop() {
-  #ifdef SCIENCE
-  science_loop();
-  #endif
-  #ifdef POWER
-  power_loop();
-  #endif
-  #ifdef SENSOR
-  sensor_loop();
-  #endif
-  #ifdef UPPER_ARM
-  upper_arm_loop();
-  #endif
-  #ifdef LOWER_ARM
-  lower_arm_loop();
-  #endif
-  #ifdef DRIVE
-  drive_loop();
-  #endif
+  if(tflag == 1){
```

Comment:
The runtime program shouldn't wait for a new cycle to start to begin processing. We should have it running always doing processes, the timer should begin tasks like sending a feedback packet to the cpu or reading one. We should always be reading our sensors and making better state estimates since those will be more or less frequent and not fixed to the control cycle time.